src/data_api.rs

```diff
@@ -0,0 +1,325 @@
+//! This module contains two macros: [`impl_data_api!`] and [`impl_iterators!`].
+//! The first takes in the name of a quadrature rule struct that has a field named `node_weight_pairs`
+//! of the type `Vec<(f64, f64)>`. It also needs the names that it should give the various structs
```

Comment:
Or should it be rather Vec<(Node, Weight)> ?

Reply:
Makes sense, yes